mc admin heal crashes

```
mc admin heal play/notabene
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x461425]

goroutine 1 [running]:
panic(0x83e0a0, 0xc420012140)
	/home/harsha/.gimme/versions/go1.7.5.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/minio/mc/cmd.mainAdminHeal(0xc420021e40, 0x8c8413, 0x190)
	/home/harsha/mygo/src/github.com/minio/mc/cmd/admin-heal.go:164 +0x3c5
github.com/minio/mc/vendor/github.com/minio/cli.HandleAction(0x829b40, 0x8f6850, 0xc420021e40, 0x0, 0x0)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/app.go:497 +0xd4
github.com/minio/mc/vendor/github.com/minio/cli.(*App).RunAsSubcommand(0xc4202109c0, 0xc420021ce0, 0x0, 0x0)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/app.go:391 +0xaa8
github.com/minio/mc/vendor/github.com/minio/cli.Command.startApp(0x8ae745, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8b478f, 0x12, 0x0, ...)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/command.go:302 +0x8e5
github.com/minio/mc/vendor/github.com/minio/cli.Command.Run(0x8ae745, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8b478f, 0x12, 0x0, ...)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/command.go:100 +0x16a9
github.com/minio/mc/vendor/github.com/minio/cli.(*App).RunAsSubcommand(0xc420210680, 0xc420020f20, 0x0, 0x0)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/app.go:386 +0xb1e
github.com/minio/mc/vendor/github.com/minio/cli.Command.startApp(0x8aebde, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8b60be, 0x15, 0x0, ...)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/command.go:302 +0x8e5
github.com/minio/mc/vendor/github.com/minio/cli.Command.Run(0x8aebde, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8b60be, 0x15, 0x0, ...)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/command.go:100 +0x16a9
github.com/minio/mc/vendor/github.com/minio/cli.(*App).Run(0xc4200876c0, 0xc42000cc00, 0x4, 0x4, 0x0, 0x0)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/app.go:260 +0x816
github.com/minio/mc/vendor/github.com/minio/cli.(*App).RunAndExitOnError(0xc4200876c0)
	/home/harsha/mygo/src/github.com/minio/mc/vendor/github.com/minio/cli/app.go:281 +0x53
github.com/minio/mc/cmd.Main()
	/home/harsha/mygo/src/github.com/minio/mc/cmd/main.go:90 +0x163
main.main()
	/home/harsha/mygo/src/github.com/minio/mc/main.go:28 +0x14
```

## Expected behaviour
Does not crash

## Actual behaviour
Crashes

## Steps to reproduce the behaviour
Do not know

## mc version
- (paste output of `mc version`)

## System information

--- a/cmd/admin-heal-list.go
+++ b/cmd/admin-heal-list.go
@@ -205,6 +205,10 @@ func mainAdminHealList(ctx *cli.Context) error {
 				errorIf(probe.NewError(obj.Err), "Cannot heal object `"+obj.Key+"`.")
 				continue
 			}
+			// Skip for non-recursive use case.
+			if obj.HealObjectInfo == nil {
+				continue
+			}
 			printMsg(healListMessage{Bucket: currBucket, Object: obj})
 		}
 	}

--- a/cmd/admin-heal.go
+++ b/cmd/admin-heal.go
@@ -159,6 +159,10 @@ func mainAdminHeal(ctx *cli.Context) error {
 			errorIf(probe.NewError(obj.Err), "Cannot list objects to be healed.")
 			continue
 		}
+		// Heal object info is nil skip it, must be a directory.
+		if obj.HealObjectInfo == nil {
+			continue
+		}
 
 		// Check the heal status, and call heal object API only when an object can be healed
 		switch healInfo := *obj.HealObjectInfo; healInfo.Status {

